crates/turborepo-lib/src/turbo_json/mod.rs

```diff
@@ -565,21 +604,38 @@ type TurboJSONValidation = fn(&TurboJson) -> Vec<Error>;
 pub fn validate_no_package_task_syntax(turbo_json: &TurboJson) -> Vec<Error> {
     turbo_json
         .pipeline
-        .keys()
-        .filter(|task_name| task_name.is_package_task())
-        .map(|task_name| Error::UnnecessaryPackageTaskSyntax {
-            actual: task_name.to_string(),
-            wanted: task_name.task().to_string(),
+        .iter()
+        .filter(|(task_name, _)| task_name.is_package_task())
+        .map(|(task_name, entry)| {
+            let (span, text) = match (entry.span.clone(), &turbo_json.text) {
+                (Some(range), Some(text)) => (Some(range.into()), text.to_string()),
+                (_, _) => (None, String::new()),
+            };
```

Comment:
This pattern seems common enough that we might wanna put it in a helper function